config/condition_nodes.go

```diff
@@ -0,0 +1,88 @@
+package config
+
+import "fmt"
+
+const nodesConditionTypes = "nodes"
+
+type NodesConditionConfig struct {
+	Datacenter *string `mapstructure:"datacenter"`
```

Comment:
I see that the query parameter `Filter` is available and supported in the [node template](https://github.com/hashicorp/consul-terraform-sync/pull/358/files#diff-bedd4bfa666cdb3e010c470094d77cfe00ecf368feeb37a03a2a97042b59bfacR206-R209), it would be a good feature to expose through the configuration to be used.

I like the omission of the `node-meta` parameter since it's marked as [deprecated for the `/v1/catalog/nodes` endpoint](https://www.consul.io/api-docs/catalog#node-meta)